File: Programmers/python/전화번호 목록.py
# 주의점: 이중 for문은 O(n^2)로 효율성 테스트에서 떨어집니다.  

def solution(phone_book):  
    phone_book.sort()
    
    for idx in range(len(phone_book) - 1):
        prefix = phone_book[idx]
        prefix_len = len(prefix)
        target = phone_book[idx + 1]
        
        if prefix == target[:prefix_len]:
            return False
    
    return True

File: Programmers/java/전화번호 목록.java
import java.util.Arrays;

class Solution {
    public boolean solution(String[] phone_book) {
        boolean answer = true;
        Arrays.sort(phone_book);
        
        for(int i = 0; i < phone_book.length - 1; i++) {
            String target = phone_book[i];
            String compare = phone_book[i+1];
            if(compare.startsWith(target)) {
                return false;
            }
        }
        
        return answer;
    }
}
